src/base/rss/rssfolder.cpp

```diff
@@ -189,7 +189,6 @@ QHash<QString, FeedPtr> Folder::getAllFeedsAsHash() const
 
 bool Folder::addFile(const FilePtr &item)
 {
-    Q_ASSERT(!m_children.contains(item->id()));
```

Comment:
But we should not have two files (feeds or folders) with the same ID here!
I have not yet checked your changes completely but I still think it looks like a mistake.